game/player.go

```diff
@@ -29,10 +29,12 @@ func NewPlayer(deck *Deck) *Player {
 }
 
 func (p *Player) Draw() {
-	p.AddToHand(p.Deck.Draw())
+	card := p.Deck.Draw()
+	p.AddToHand(card)
 }
 
 func (p *Player) AddToHand(c *Card) {
+	// TODO this seems like suspicious code, why would a card be nil here?
```

Comment:
A card would be nil here if you attempted to start a game with less than 7 cards in your deck. :P